spring-cloud-commons/src/main/java/org/springframework/cloud/client/circuitbreaker/CircuitBreakerBuilder.java

```diff
@@ -0,0 +1,13 @@
+package org.springframework.cloud.client.circuitbreaker;
+
+/**
+ * Constructs circuit breakers.
+ *
+ * @author Ryan Baxter
+ */
+public interface CircuitBreakerBuilder {
+
+	public CircuitBreakerBuilder id(String id);
```

Comment:
There are ways to make the return type generic so you can reference methods specific to the implementation.

Reply:
You mean on the `build` method?

Reply:
I'm kind of confused because you are working with the abstraction not the underlying implementation, thats the point

Reply:
Your right. https://github.com/spring-cloud/spring-cloud-netflix/pull/3277/files#diff-c7ed37301464ccebbbb9c6efb0787d36R26 it works.